src/zjs_i2c.c

```diff
@@ -279,18 +257,30 @@ static jerry_value_t zjs_i2c_open(const jerry_value_t function_obj,
         return zjs_error("zjs_i2c_open: missing required field (speed)");
     }
 
-    // send IPM message to the ARC side
-    zjs_ipm_message_t send;
-    zjs_ipm_message_t reply;
-
-    send.type = TYPE_I2C_OPEN;
-    send.data.i2c.bus = (uint8_t)bus;
-    send.data.i2c.speed = (uint8_t)speed;
-
-    bool success = zjs_i2c_ipm_send_sync(&send, &reply);
-
-    if (!success) {
-        return zjs_error("zjs_i2c_write: ipm message failed or timed out!");
+    if (bus < MAX_I2C_BUS) {
+        char i2c_bus[6];
+        snprintf(i2c_bus, 6, "I2C_%i", (uint8_t)bus);
+        i2c_device[bus] = device_get_binding(i2c_bus);
+        if (!i2c_device[bus]) {
+            ZJS_PRINT("I2C bus %s not found.\n", i2c_bus);
```

Comment:
This error could probably be combined with the "not a valid bus" one below, combine these two if's - like above.